# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/ViewportFrame.cpp
# Commit: a261bb633790
# Full Hash: a261bb633790ddca9cc8ce83adca0a1bb36114a6
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2022-02-23 09:38:00
# Regressor Bug: 1714584
# File Overlap Count: 2
# Description:
#   Bug 1714584 - Part 1: Decouple nsDisplayList internal list from nsDisplayItems r=mstange
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138152
# ==============================================================================

diff -r 635f402efb1b -r a261bb633790 layout/generic/ViewportFrame.cpp
--- a/layout/generic/ViewportFrame.cpp	Tue Feb 22 23:42:08 2022 +0000
+++ b/layout/generic/ViewportFrame.cpp	Tue Feb 22 23:42:18 2022 +0000
@@ -124,7 +124,7 @@
   nsDisplayListBuilder::AutoBuildingDisplayList buildingForChild(
       aBuilder, aFrame, visible, dirty);
 
-  nsDisplayList list;
+  nsDisplayList list(aBuilder);
   aFrame->BuildDisplayListForStackingContext(aBuilder, &list);
   aList->AppendToTop(&list);
 }
@@ -135,7 +135,7 @@
   // The common case for ::backdrop elements on the top layer is a single
   // fixed position container, holding an opaque background color covering
   // the whole viewport.
-  if (aList->Count() != 1 ||
+  if (aList->Length() != 1 ||
       aList->GetTop()->GetType() != DisplayItemType::TYPE_FIXED_POSITION) {
     return false;
   }
@@ -168,7 +168,7 @@
 
 nsDisplayWrapList* ViewportFrame::BuildDisplayListForTopLayer(
     nsDisplayListBuilder* aBuilder, bool* aIsOpaque) {
-  nsDisplayList topLayerList;
+  nsDisplayList topLayerList(aBuilder);
 
   nsTArray<dom::Element*> topLayer = PresContext()->Document()->GetTopLayer();
   for (dom::Element* elem : topLayer) {